app/src/main/java/com/github/se/cyrcle/ui/review/ReviewScreen.kt

```diff
@@ -0,0 +1,125 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.review.ReviewViewModel
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.navigation.Screen
+import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReviewScreen(
+    navigationActions: NavigationActions,
+    parkingViewModel: ParkingViewModel,
+    reviewViewModel: ReviewViewModel
+) {
+  var sliderValue by remember { mutableStateOf(0f) }
+  var textValue by remember { mutableStateOf("") }
+
+  val context = LocalContext.current // Get the current context
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            navigationActions = navigationActions,
+            title = {
+              Text(
+                  text = "Add Your Review",
+                  fontSize = 20.sp,
+                  fontWeight = FontWeight.Bold,
+                  color = MaterialTheme.colorScheme.secondary,
+                  modifier = Modifier.testTag("TopAppBarTitle"))
+            })
+      }) { paddingValues ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(paddingValues).padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center) {
+              // Stars reflecting the slider value (supports half stars)
+              Row(
+                  modifier = Modifier.padding(8.dp).testTag("StarsRow"),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    for (i in 1..5) {
+                      val starIcon =
+                          when {
+                            i <= sliderValue -> Icons.Filled.Star // Full star
+                            i - 0.5f == sliderValue -> Icons.Filled.StarHalf // Half star
+                            else -> Icons.Filled.StarBorder // Empty star
+                          }
+                      Icon(
+                          imageVector = starIcon,
+                          contentDescription = null,
+                          tint = MaterialTheme.colorScheme.primary,
+                          modifier = Modifier.size(40.dp).testTag("Star$i"))
+                    }
+                  }
+
+              // Display the experience text based on the slider value
+              Text(
+                  text =
+                      when (sliderValue) {
+                        in 0f..1f -> "Terrible experience!"
+                        in 1f..2f -> "Bad experience :("
+                        in 2f..3f -> "Poor experience :/"
+                        3f -> "Average experience..."
+                        in 3.5f..4f -> "Good experience!"
+                        in 4.5f..5f -> "Great experience!"
+                        else -> ""
+                      },
```

Comment:
I'm not 100% onboard with the smileys, especially as there are not consistent along all lines (some have 1, some have none, and the `...` after `Average experience` might convey a "sarcasm" vibes)

Reply:
I'd like to integrate some actual emojis down the line (I figured it'd create some sort of visual cue to homogenize which "feeling" maps to what score) , so these are just meant to be placeholders. I'll note this though, I'll bring up the general UI of this screen at the scrum meeting